doc: bluetooth: mesh: rework DFU over Bluetooth Mesh guide and samples (#30086)
nrfconnect/sdk-nrf: samples/bluetooth/mesh/dfu/distributor/README.rst

Performing a Device Firmware Update
-----------------------------------

The end-to-end procedure for performing a Device Firmware Update over Bluetooth Mesh (uploading the image to the Distributor, populating the receivers list, initiating the distribution, and applying the image on the Target nodes) is described in :ref:`dfu_over_bt_mesh`.

> Follow the steps in that guide to run a DFU with this sample.

Reviewer: ```suggestion
Follow the steps in the guide to run a DFU with this sample.
```

> The Bluetooth Mesh DFU Model specification defines the Firmware Update Initiator role to control the firmware distribution.

Reviewer: ```suggestion
The Bluetooth Mesh DFU Model specification defines the Firmware Update Initiator's role of controlling the firmware distribution.
```

This sample supports, but does not require, an external Initiator to control the DFU procedure.
You can drive the procedure in the following ways:

* Using the nRF Mesh mobile app as an Initiator (recommended, and covered step-by-step in :ref:`dfu_over_bt_mesh`).
* Using the shell commands provided by the Bluetooth Mesh DFU subsystem in Zephyr instead of an Initiator.

>   The shell commands can be executed either through the shell management subsystem of MCU manager (for example, using the `nRF Connect Device Manager`_ application, or :ref:`Mcumgr command-line tool <dfu_tools_mcumgr_cli>`), or using the :ref:`zephyr:shell_api` module over RTT.

Reviewer: ```suggestion
  You can execute the shell commands using either the shell management subsystem of the MCU manager (for example, using the `nRF Connect Device Manager`_ application, or :ref:`Mcumgr command-line tool <dfu_tools_mcumgr_cli>`), or the :ref:`zephyr:shell_api` module over RTT.
```

To activate the new firmware image on the Target node, the image must satisfy the acceptance criteria for the target (for example, see the Target sample requirements described in :ref:`ble_mesh_dfu_target_upgrade`).

Expected distribution duration
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The time required to distribute a firmware image to the Target nodes depends on the number of Target nodes, the Transfer Mode (multicast versus unicast), the image size, the network topology and the network quality.

Reviewer: ```suggestion
The time required to distribute a firmware image to the Target nodes depends on the following factors:

* The number of Target nodes
* the Transfer Mode (multicast or unicast)
* The image size
* The network topology
* The network quality
```
This might improve readability.

The following figures are indicative reference numbers observed with the |NCS| Bluetooth Mesh DFU samples and are provided as an order-of-magnitude estimate only.

.. table::
   :align: left

   +----------------+---------------------+-----------------------+
   | Transfer Mode  | Small deployment    | Large deployment      |
   +================+=====================+=======================+
   | Multicast      | ~2 hours (2 nodes)  | ~2 h 15 min           |
   |                |                     | (~135 min, 100 nodes) |
   +----------------+---------------------+-----------------------+
   | Unicast        | ~34 minutes         | ~3400 minutes         |
   |                | (1 node)            | (~57 h, 100 nodes)    |
   +----------------+---------------------+-----------------------+

> In multicast transfer mode, the Distributor sends the image once and all receivers pick it up in parallel, so the total duration grows only moderately with the number of Target nodes.

Reviewer: transfer mode or Transfer Mode? You have it with initial caps above. Is it a spec term?

Author: changed on distribution addressing

In unicast transfer mode, the Distributor sends the image sequentially to each Target node, so the total duration grows roughly linearly with the number of Target nodes.

.. _ble_mesh_dfu_distributor_fw_image_upload:

Alternative ways of uploading a firmware image
----------------------------------------------

The main procedure for uploading a firmware image to the Distributor is the phone-based flow described in :ref:`dfu_over_bt_mesh`.

> The alternatives documented in this section can be used when the phone-based flow is not applicable, for example in automated setups or when the image is provided by another Initiator device on the mesh network.

Reviewer: ```suggestion
You can use the alternatives documented in this section when the phone-based flow is not applicable, for example, in automated setups or when the image is provided by another Initiator device on the mesh network.
```

You can drive the procedure in the following ways:

* Using the nRF Mesh mobile app as an Initiator (recommended, and covered step-by-step in :ref:`dfu_over_bt_mesh`).
* Using the shell commands provided by the Bluetooth Mesh DFU subsystem in Zephyr instead of an Initiator.
  You can execute the shell commands using either the shell management subsystem of the MCU manager (for example, using the `nRF Connect Device Manager`_ application, or :ref:`Mcumgr command-line tool <dfu_tools_mcumgr_cli>`), or the :ref:`zephyr:shell_api` module over RTT.

To activate the new firmware image on the Target node, the image must satisfy the acceptance criteria for the target (for example, see the Target sample requirements described in :ref:`ble_mesh_dfu_target_upgrade`).

Expected distribution duration
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The time required to distribute a firmware image to the Target nodes depends on the following factors:

* The number of Target nodes
* The distribution addressing (multicast group address versus per-target unicast address)
* The image size
* The network topology
* The network quality

> The following figures are indicative reference numbers observed with the |NCS| Bluetooth Mesh DFU samples and are provided as an order-of-magnitude estimate only.

Reviewer 2: Add compiled binary size information in the table header or in the table caption
```suggestion
The following figures are indicative reference numbers observed with the |NCS| Bluetooth Mesh DFU samples (target sample build: XY KB) and are provided as an order-of-magnitude estimate only.
```

.. table::
   :align: left

   +----------------+---------------------+-----------------------+
   | Distribution   | Small deployment    | Large deployment      |
   | addressing     |                     |                       |
   +================+=====================+=======================+
   | Multicast      | ~2 hours (2 nodes)  | ~2 h 15 min           |
   |                |                     | (~135 min, 100 nodes) |
   +----------------+---------------------+-----------------------+
   | Unicast        | ~34 minutes         | ~3400 minutes         |
   |                | (1 node)            | (~57 h, 100 nodes)    |

>    +----------------+---------------------+-----------------------+

Reviewer 2: This data will have to be cross-checked with CI data for unicast updates.

Author: CI shows: [2061.530](https://jenkins-ncs.nordicsemi.no/job/latest/job/night/job/test-fw-nrfconnect-ble_mesh/job/master/1381/testReport/mesh.samples.dfu_upload.system_tests)
This is ~34 minutes (1 node)
